src/frontend/src/optimizer/plan_node/logical_source.rs

```diff
@@ -83,6 +84,76 @@ impl LogicalSource {
         }
     }
 
+    pub fn gen_optional_generated_column_project_exprs(
```

Comment:
IIUC, we do not support generated columns of `Source` in this PR?

Reply:
I haven't tested against `Source`. I baned it for now.